src/other/PropertyEditor.js

```diff
@@ -494,23 +494,46 @@
                 ;
                 break;
             default:
-
-                cell.append("input")
-                    .attr("id", this.id() + "_" + param.id)
-                    .classed("property-input", true)
-                    .on("change", function () {
-                        context.setProperty(widget, param.id, this.value);
-                    })
-                ;
-                if (param.type === "html-color" && !this.isIE) {
+                if(param.ext && param.ext.range){
+                    cell.append("span")
+                        .classed("property-input-span", true)
+                        .attr("id", this.id() + "_" + param.id + "_currentVal")
+                        .text(param.defaultValue)
+                    ;
+                    cell.append("input")
+                        .attr("type", "range")
+                        .attr("step", param.ext.range.step)
+                        .attr("min", param.ext.range.min)
+                        .attr("max", param.ext.range.max)
+                        .attr("id", this.id() + "_" + param.id)
+                        .classed("property-input", true)
+                        .on("input", function () {
+                            context.setProperty(widget, param.id, this.value);
+                            d3.select('#'+this.id+"_currentVal").text('Current Value: '+this.value);
+                        })
+                        .on("change", function () {
+                            context.setProperty(widget, param.id, this.value);
+                            d3.select('#'+this.id+"_currentVal").text('Current Value: '+this.value);
+                        })
+                    ;
+                }else{
                     cell.append("input")
-                        .attr("id", this.id() + "_" + param.id + "_2")
+                        .attr("id", this.id() + "_" + param.id)
                         .classed("property-input", true)
-                        .attr("type", "color")
                         .on("change", function () {
                             context.setProperty(widget, param.id, this.value);
                         })
                     ;
+                    if (param.type === "html-color" && !this.isIE) {
+                        cell.append("input")
+                            .attr("id", this.id() + "_" + param.id + "_2")
+                            .classed("property-input", true)
+                            .attr("type", "color")
+                            .on("change", function () {
+                                context.setProperty(widget, param.id, this.value);
+                            })
+                        ;
+                    }
```

Comment:
@jbrundage  I thought we had color already - but I guess not?

Reply:
Github's diff is a bit misleading. That color conditional was on line 505 in the old file.

I believe this seems new because of the indentation?